inc/saiarsqualitymap.h

```diff
@@ -0,0 +1,353 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiarsqualitymap.h
+ *
+ * @brief   This module defines SAI QOS Maps interface
+ */
+
+#if !defined (__SAIARSQUALITYMAP_H_)
+#define __SAIARSQUALITYMAP_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIARSQUALITYMAP SAI - ARS quality map specific API definitions.
+ *
+ * @{
+ */
+
+/**
+ * @brief Adaptive Routing and Switching quality measure algorithm
+ */
+typedef enum _sai_ars_quality_map_algo_t
+{
+    /** Exponentially weighted moving average */
+    SAI_ARS_QUALITY_MAP_ALGO_EWMA,
+
+} sai_ars_quality_map_algo_t;
+
+/**
```

Comment:
Where is this sai_ars_quality_map_algo_t used/referenced ?

Reply:
Quality map object has a switch binding

Reply:
Wondering whether the algo ID (SAI_ARS_QUALITY_MAP_ALGO_EWMA) is used. Don't see "sai_ars_quality_map_algo_t" is referred by some object

Reply:
What if another algo with its own set of configuration parameters is chosen by a vendor  ? How can it be modelled ?

Reply:
> Where is this sai_ars_quality_map_algo_t used/referenced ?

Its a per switch configuration. Added it to the ars qualtymap object.

Reply:
> ARS Quality Map seems to be very algo-specific. Why are some params put into Switch object and some - into ARS Quality Map singleton ?

Moved all the ARS related parameters from switch to ARS quality map object. Now the NOS has to first check if ARS quality map object is supported using capability API and then can query the attribute for max scale support for a given HW.